jd4/cgroup.py

```diff
@@ -10,58 +10,63 @@
 from jd4.log import logger
 from jd4.util import read_text_file, write_text_file
 
-CPUACCT_CGROUP_ROOT = '/sys/fs/cgroup/cpuacct/jd4'
-MEMORY_CGROUP_ROOT = '/sys/fs/cgroup/memory/jd4'
-PIDS_CGROUP_ROOT = '/sys/fs/cgroup/pids/jd4'
+CGROUP2_ROOT = '/sys/fs/cgroup/jd4'
+CGROUP2_DAEMON_ROOT = path.join(CGROUP2_ROOT, 'daemon')
 WAIT_JITTER_NS = 5000000
 
 def try_init_cgroup():
     euid = geteuid()
-    cgroups_to_init = list()
-    if not (path.isdir(CPUACCT_CGROUP_ROOT) and access(CPUACCT_CGROUP_ROOT, W_OK)):
-        cgroups_to_init.append(CPUACCT_CGROUP_ROOT)
-    if not (path.isdir(MEMORY_CGROUP_ROOT) and access(MEMORY_CGROUP_ROOT, W_OK)):
-        cgroups_to_init.append(MEMORY_CGROUP_ROOT)
-    if not (path.isdir(PIDS_CGROUP_ROOT) and access(PIDS_CGROUP_ROOT, W_OK)):
-        cgroups_to_init.append(PIDS_CGROUP_ROOT)
-    if cgroups_to_init:
+    if not (path.isdir(CGROUP2_ROOT) and access(CGROUP2_ROOT, W_OK)):
```

Comment:
if ... return to simplify control flow

Reply:
We cannot return here because we should do the below things (`Put myself into the cgroup that I can write.`) each time the process gets started.